src/AccessibilityInsights/Modes/ConfigurationModeControl.xaml.cs

```diff
@@ -141,22 +143,38 @@ private void buttonOk_Click(object sender, RoutedEventArgs e)
                 MainWin.UpdateMainWindowConnectionFields();
             }
 
-            if (appSettingsCtrl.SelectedReleaseChannel != Configuration.ReleaseChannel)
+            if (await HandleReleaseSwitch().ConfigureAwait(false))
+                return;
+
+            Dispatcher.Invoke(() => MainWin.TransitionToSelectActionMode());
+        }
+
+        private async Task<bool> HandleReleaseSwitch()
+        {
+            var channelDialog = new ChangeChannelContainedDialog(appSettingsCtrl.SelectedReleaseChannel);
+
+            if (appSettingsCtrl.SelectedReleaseChannel == Configuration.ReleaseChannel ||
+                !await MainWin.ctrlDialogContainer.ShowControl(channelDialog).ConfigureAwait(false))
             {
-                try
-                {
-                    VersionSwitcherWrapper.ChangeChannel(appSettingsCtrl.SelectedReleaseChannel);
-                    MainWin.Close();
-                    return;
-                }
-                catch (Exception ex)
-                {
-                    ex.ReportException();
-                    MessageDialog.Show(string.Format(CultureInfo.InvariantCulture, Properties.Resources.ConfigurationModeControl_VersionSwitcherException, ex.Message));
-                }
+                return false;
+            }
+
+            try
+            {
+                VersionSwitcherWrapper.ChangeChannel(appSettingsCtrl.SelectedReleaseChannel);
+                Dispatcher.Invoke(() => MainWin.Close());
+                return true;
+            }
+            catch (Exception ex)
+            {
+                ex.ReportException();
+
+                Dispatcher.Invoke(() =>
+                    MessageDialog.Show(string.Format(CultureInfo.InvariantCulture, Properties.Resources.ConfigurationModeControl_VersionSwitcherException, ex.Message))
```

Comment:
>ex.Message [](start = 155, length = 10)

As we discussed, using the raw exception text here (at least when I have no network) is neither actionable nor pretty. We've already reported the Exception via telemetry, so let's display a message here that is more applicable to the user

Reply:
Message updated

Reply:
Message updated